core/retries/src/main/java/software/amazon/awssdk/retries/LegacyRetryStrategy.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.retries;
+
+import java.util.function.Predicate;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+import software.amazon.awssdk.retries.api.BackoffStrategy;
+import software.amazon.awssdk.retries.api.RetryStrategy;
+import software.amazon.awssdk.retries.internal.LegacyRetryStrategyImpl;
+import software.amazon.awssdk.retries.internal.circuitbreaker.TokenBucketStore;
+
+/**
+ * The legacy retry strategy is a {@link RetryStrategy} for normal use-cases.
+ * <p>
+ * Like the {@link StandardRetryStrategy}, the legacy strategy is generally useful across all retry use-cases. Unlike the
+ * {@link StandardRetryStrategy} this strategy has special treatment for throttling exceptions by being less sensitive to
+ * triggering the circuit breaking and also by backing off with a higher base delay.
+ * <p>
+ * Unlike {@link AdaptiveRetryStrategy}, the legacy strategy is generally useful across all retry use-cases.
+ * <p>
+ * The legacy retry strategy by default:
+ * <ol>
+ *     <li>Retries on the conditions configured in the {@link Builder}.
+ *     <li>Retries 2 times (3 total attempts). Adjust with {@link Builder#maxAttempts(int)}
+ *     <li>For non-throttling exceptions uses the {@link BackoffStrategy#exponentialDelay} backoff strategy, with a base delay
+ *     of 100 milliseconds and max delay of 20 seconds. Adjust with {@link Builder#backoffStrategy}
+ *     <li>For throttling exceptions uses the {@link BackoffStrategy#exponentialDelay} backoff strategy, with a base delay of
+ *     500 milliseconds  and max delay of 20 seconds. Adjust with {@link LegacyRetryStrategy.Builder#throttlingBackoffStrategy}
+ *     <li>Circuit breaking (disabling retries) in the event of high downstream failures across the scope of
+ *     the strategy. The circuit breaking will never prevent a successful first attempt. Adjust with
+ *     {@link Builder#circuitBreakerEnabled}.
```

Comment:
IIRC, in the existing legacy implementation, retrying throttling error doesn't consume token. We should probably mention it here as well.

Reply:
I will add a list item saying "The state of the circuit breaker is not affected by throttling exceptions", The way we implement the circuit breaker logic is an implementation detail that we don't expose in the interface so I think is better if we don't mention that here.